engine/app/components/citizens_advice_components/radio_group.rb

```diff
@@ -0,0 +1,93 @@
+# frozen_string_literal: true
+
+module CitizensAdviceComponents
+  class RadioGroup < Base
+    attr_reader :legend, :name, :error_message, :hint, :size, :layout
+
+    renders_many :radio_buttons, "RadioButton"
+
+    def initialize(legend:, name:, options: nil)
+      super
+      @legend = legend
+      @name = name
+
+      set_options(options)
+    end
+
+    def set_options(options)
+      return unless options.present?
+
+      @error_message = options[:error_message]
+      @hint = options[:hint]
+      @optional = fetch_or_fallback_boolean(options[:optional], fallback: false)
+      @size = fetch_or_fallback(given_value: options[:size], allowed_values: [nil, :regular, :small], fallback: :regular)
+      @layout = fetch_or_fallback(given_value: options[:layout], allowed_values: [nil, :list, :inline], fallback: :list)
+    end
+
+    def render?
+      radio_buttons.present?
+    end
+
+    def error?
+      @error_message.present?
+    end
+
+    def optional?
+      @optional.present?
+    end
+
+    def size?
+      @size.present?
+    end
+
+    def layout?
+      @layout.present?
+    end
+
+    def hint?
+      @hint.present?
+    end
+
+    def fieldset_attributes
+      {
+        class: [
+          ("cads-radio-group--#{@size.to_s.dasherize}" if size?),
+          ("cads-radio-group--#{@layout.to_s.dasherize}" if layout?)
+        ]
+      }
+    end
+
+    class RadioButton < Base
+      attr_reader :label
+
+      def initialize(label:, value:, checked: false, **additional_attributes)
+        super
+        @label = label
```

Comment:
This is where I had accidentally added commas.  So `label` was an array of all the arguments, but all the arguments also had their own values.  It was a fun time.